modules/statistics/php/NDB_Form_statistics.class.inc

```diff
@@ -1029,6 +1060,49 @@ class NDB_Form_Statistics extends NDB_Form
             $Param_Project ='';
         }
 
+        //SUBPROJECTS
+        $subprojects[null] = 'All Valid Sub-Projects';
+
+        foreach (Utility::getSubprojectsForProject($currentProject)
+            as $key=>$value) {
+            $subprojects[$key] = $value;
+        }
+
+        foreach ($subprojects as $key=>$val) {
+            $subprojList .= $key.",";
+        }
+
+        $this->tpl_data['Subprojects'] = $subprojects;
+        $currentSubProject = null;
+        $subprojList       = substr($subprojList, 1, -1);
+        if (isset($_REQUEST['ReliabilitySubproject'])
+            && $_REQUEST['ReliabilitySubproject'] != ''
+            && in_array($_REQUEST['ReliabilitySubproject'], array_keys($subprojects))
+        ) {
+            $currentSubProject = $_REQUEST['ReliabilitySubproject'];
+            $this->tpl_data['CurrentSubproject']
+                = array(
+                   'ID'   => $currentSubProject,
+                   'Name' => $subprojects[$currentSubProject],
+                  );
+            $subproject_query     = "AND (s.SubprojectID IS NULL ";
+            $subproject_query    .= "OR s.SubprojectID=:spid)";
+            $this->params['spid'] =$currentSubProject;
+        } else {
+            $subproject_query ="";
+        }
+        if (!in_array(
+            $_REQUEST['BehaviouralSubproject'],
+            array_keys($subprojects)
+        )) {
+            $BS = 'BehaviouralSubproject';
+            $this->tpl_data['PreviousSubproject']
+                = Utility::getSubprojectsForProject()[$_REQUEST[$BS]];
```

Comment:
maybe I'm missing something that defends against it here the same way I missed in_array above, but this looks like it's vulnerable to an XSS attack to me. The guard on this if statement is explicitly if it's _not_ in subprojects, so it doesn't protect malicious input from getting here.

(The value assigned to the template should pass through something that escapes it like htmlspecialchars to fix it.)